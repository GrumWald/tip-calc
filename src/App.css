html, body {
    margin: 0;
    padding: 0;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    background-color: rgb(163, 194, 214);
    color: white;
    box-sizing: border-box;
}

header {
    height: 100px;
    background-color: rgb(51, 50, 50);   
}

header h1 {
    margin:0;
    line-height: 100px;
    margin-left: 2rem;
    font-size: 50px;
    letter-spacing: 0.2rem;
    
}

.tip-form {
    height: 650px;
    width: 500px;
    background-color: rgb(51, 50, 50);
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 2rem;
    box-sizing: border-box;
    -webkit-box-shadow: 10px 10px 25px 0px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 10px 10px 25px 0px rgba(0, 0, 0, 0.75);
    box-shadow: 10px 10px 25px 0px rgba(0, 0, 0, 0.75);
}

label {
    display: block;
    margin-top: 1.5rem;
    font-size: 20px;

}

.form-input, select {
    box-sizing: border-box;
    left: 50%;
    display: block;
    height: 40px;
    width: 350px;
    padding: 5px 10px;
    margin-top: 0.5rem;
    font-size: 20px;
    text-align: center;
}

select {
    text-align-last:center;
}

input[type='submit'] {
    height: 70px;
    width: 250px;
    margin-top: 2rem;
    margin-bottom: 2rem;
    font-size: 25px;
    background-color:rgba(55, 168, 55, 0.534);
    color: white;
    border: 1px solid white;
    border-radius: 10px;
}

input[type='submit']:hover {
    background-color: rgb(55, 168, 55);
    transition: 0.5s;
}

.result {
    margin-top: 1rem;
    height: 60px;
    line-height: 60px;
    width: 400px;
    border: 1px solid white;
    font-size: 35px;
    text-align: center;
    border-radius: 10px;
}


