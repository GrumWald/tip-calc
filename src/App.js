import React from 'react';

import TipForm from './components/TipForm';
import './App.css';

function App() {
  return (
    <div className="App">
        <header>
            <h1>TIP-CALC</h1>      
        </header>
        <TipForm />
    </div>
  );
}

export default App;
