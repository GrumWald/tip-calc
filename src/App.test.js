import React from 'react';
import { configure, shallow } from 'enzyme';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure( { adapter: new Adapter() } );

it( "renders without crashing", () => {
    shallow( <App /> );
} );

it( "renders TipCalc header", () => {
    const wrapper = shallow( <App /> );
    const header = <h1>TIP-CALC</h1>;
    expect( wrapper.contains( header ) ).toEqual( true );
} );

it( " renders correctly", () => {
    const tree = shallow( <App /> );
    expect( toJson( tree ) ).toMatchSnapshot();
} );