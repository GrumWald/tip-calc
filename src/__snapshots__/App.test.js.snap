// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      TIP-CALC
    </h1>
  </header>
  <TipForm />
</div>
`;
